src/Workspaces/Core/Portable/Diagnostics/DiagnosticData.cs

```diff
@@ -351,6 +351,11 @@ public static DiagnosticData Create(Workspace workspace, Diagnostic diagnostic)
 
         public static DiagnosticData Create(Project project, Diagnostic diagnostic)
         {
+            return Create(project, diagnostic, isSuppressed: diagnostic.IsSuppressed);
+        }
+
+        public static DiagnosticData Create(Project project, Diagnostic diagnostic, bool isSuppressed)
```

Comment:
this API looks a bit weird since Diagnostic and isSuppressed given can be different. if this is for the host diagnostics, why are they go through diagnostic <-> diagnostic data conversion rather than start from diagnostic data?